dev_tool/src/jumpbox.ts

```diff
@@ -0,0 +1,323 @@
+                console.error('Error describing instance', instance)
+                return instance
+            }
+
+            const state = instance.State?.Code
+
+            if (state === undefined) {
+                return new Error('didnt get state back for the instance')
+            }
+
+            process.stdout.write('.')
+
+            if (state === instanceStateCode) {
+                return true
+            }
+        } catch (err) {
+            return err
+        }
+
+        return false
+    }, 60 * 1000) // 60 second timeout
+    if (stoppedResult instanceof Error) {
+        return stoppedResult
+    }
+    return undefined
+}
+
+// ensureJumpboxIsRunning gets the jumpbox Instance and if it's not running it starts it
+async function ensureJumpboxIsRunning(): Promise<Instance | Error> {
+    const instance = await describeInstance({
+        Filters: [
+            {
+                Name: 'tag:mcr-vmuse',
+                Values: ['jumpbox'],
+            },
+        ],
+    })
+    if (instance instanceof Error) {
+        console.error('error fetching jumpbox', instance)
+        return instance
+    }
+
+    const jumpboxStartInstance = instance
+
+    const jumpboxStartID = jumpboxStartInstance.InstanceId
+    const jumpboxStartState = jumpboxStartInstance.State?.Code
+
+    if (!jumpboxStartID || jumpboxStartState === undefined) {
+        return new Error(
+            `AWS didn't return info we needed. id: ${jumpboxStartID} state: ${jumpboxStartState}`
+        )
+    }
+
+    if (jumpboxStartState === 16) {
+        // running state code
+        console.info('Jumpbox is running')
+        return jumpboxStartInstance
+    }
+
+    console.info('Jumpbox is not running', jumpboxStartInstance.State?.Name)
+
+    if (jumpboxStartState !== 80) {
+        console.info('Jumpbox is not stopped yet. waiting to start it')
+        // wait for it to be running
+        const stopped = await waitForJumpboxToReachState(jumpboxStartID, 80) // 80 is stopped
+        if (stopped instanceof Error) {
+            return stopped
+        }
+        console.info('Jumpbox Stopped')
+    }
+
+    console.info('Starting Jumpbox')
+    // issue the start command
+    const startResult = await startInstance(jumpboxStartID)
+    if (startResult instanceof Error) {
+        return startResult
+    }
+
+    const started = await waitForJumpboxToReachState(jumpboxStartID, 16) // 16 is running
+    if (started instanceof Error) {
+        return started
+    }
+
+    console.info('Jumpbox Started')
+
+    const startedInstance = await describeInstance({
+        Filters: [
+            {
+                Name: 'instance-id',
+                Values: [jumpboxStartID],
+            },
+        ],
+    })
+    if (startedInstance instanceof Error) {
+        console.error('error fetching restarted jumpbox', startedInstance)
+        return startedInstance
+    }
+
+    return startedInstance
+}
+
+async function ensureWhitelistIP(
```

Comment:
Not to be that sjw, but I think people like to use "allowlist"/"blocklist" more recently.